python/cudf/udf_cpp/shim.cu

```diff
@@ -662,3 +664,15 @@ make_definition_idx(BlockIdxMax, int64, int64_t);
 make_definition_idx(BlockIdxMax, float64, double);
 #undef make_definition_idx
 }
+
+__device__ void udf_str_dtor(void* udf_str, size_t size, void* dtor_info)
```

Comment:
I wonder how obvious it is what's going on here - could it do with some explanatory comments? I know that the purpose is to cause the `udf_string` to free its data via `clear()` rather than calling `delete` on it because it was constructed with placement `new`, but I don't know if it comes across in the code in this PR.

I'm also a bit unclear why we call `clear()` rather than being explicit that we're trying to destroy the `udf_string` with something like `ptr->~udf_string()` - is this a question for @davidwendt?

Reply:
Updated to call the destructor